src/components/external-source/ExternalSourceUpdateCard.svelte

```diff
@@ -0,0 +1,159 @@
+    >
+      <LightningCharge
+        class="filter-search-icon"
+        style="color: {deleted ? 'rgba(152, 35, 35, 0.5)' : 'rgba(152, 101, 35, 0.5)'}"
+      />
+      {#if sources.length === 1 && !deleted}
+        1 new file has been uploaded
+      {:else if sources.length === 1 && deleted}
+        1 file has been deleted
+      {:else if deleted}
+        {sources.length} files have been deleted
+      {:else}
+        {sources.length} new files have been uploaded
+      {/if}
+    </div>
+    <slot name="right">
+      <button class="st-button icon fs-6" on:click={() => dispatch('dismiss')}>
+        <XIcon />
+      </button>
+    </slot>
+  </div>
+  <div style="padding: 4px 14px 12px;">
+    <div class="card--row" style="padding-top: 5px">
+      <span class="st-typography-body">
+        {!deleted
+          ? 'New files matching source types and derivation groups in the current plan:'
+          : 'Deleted files organized by source type and derivation group:'}
+      </span>
+    </div>
+    <div class="card--body st-typography-body">
+      {#each Object.keys(mappedSources) as sourceType}
+        <Collapse title={sourceType} tooltipContent={'Source Type'} defaultExpanded={true}>
+          {#each Object.keys(mappedSources[sourceType]) as derivationGroup}
+            <Collapse title={derivationGroup} tooltipContent={'Derivation Group'} defaultExpanded={true}>
+              {#each mappedSources[sourceType][derivationGroup] as source}
+                <div class="card--source-info">
+                  <p
+                    style="flex-shrink: 1; max-width: 100%; overflow: hidden; text-overflow: ellipsis; white-space: nowrap"
+                  >
+                    {source.key}
+                  </p>
+                  <p
+                    style=" color: gray;flex-shrink: 0; margin-left: auto; padding-right: 0.25rem; padding-top: 0.1rem; white-space: nowrap"
+                  >
+                    {getTimeAgo(new Date(source.change_date), new Date(), Number.MAX_SAFE_INTEGER)}
+                  </p>
+                </div>
+              {/each}
+            </Collapse>
+          {/each}
+        </Collapse>
+      {/each}
+    </div>
+    <div style="padding-top:5px">
+      <button
+        class="st-button secondary hover-fix"
+        style="border: 0px; width: 100px"
+        on:click={() => dispatch('dismiss')}>Dismiss</button
+      >
+    </div>
+    <slot />
+  </div>
+</div>
+
+<style>
+  .card {
+    background: var(--bg-color, rgba(245, 245, 245, 0.35));
+    border: 1px solid var(--border-color, rgba(152, 101, 35, 0.5));
+    border-radius: 4px;
+    color: var(--st-gray-70);
+    cursor: pointer;
+    display: flex;
+    flex-direction: column;
+    /* padding: 4px 14px 12px; */
+    text-align: left;
+  }
+
+  .card--row {
+    align-items: center;
+    display: flex;
+    justify-content: space-between;
+    min-height: 24px;
+    width: 100%;
+  }
+
+  .card--title-row {
```

Comment:
Can you add `border-top-right-radius: 4px` and `border-top-left-radius: 4px` so that the background doesn't overflow the parent border radius? Could also hide overflow on the parent but that has a chance to cause unwanted behavior.